src/ros_filter.cpp

```diff
@@ -90,6 +90,39 @@ namespace RobotLocalization
     topicSubs_.clear();
   }
 
+  template<typename T>
+  void RosFilter<T>::reset()
+  {
+    // Get rid of any initial poses (pretend we've never had a measurement)
+    initialMeasurements_.clear();
+    previousMeasurements_.clear();
+    previousMeasurementCovariances_.clear();
+
+    // Clear the measurement queue.
+    // This prevents us from immediately undoing our reset.
+    while (!measurementQueue_.empty() && ros::ok())
+    {
+      measurementQueue_.pop();
+    }
+
+    ros::getGlobalCallbackQueue()->clear();
```

Comment:
I "stole" the idea of calling getGlobalCallbackQueue from `RosFilter<T>::setPoseCallback` it mainly does something similar. a prematurely exit sounds like an error in the code somewhere. So just moving it just to the end might not be a good solution.